opensearch-operator/go.mod

```diff
@@ -4,6 +4,7 @@ go 1.16
 
 require (
 	cloud.google.com/go v0.99.0 // indirect
+	github.com/Masterminds/semver v1.5.0
```

Comment:
It's a pretty full featured package that allows us to work with SemVers and do comparisons.  I used this because it gives us a bit more flexibility in the future if we want to do some more version validations.